low-level/files/opt/seagate/sspl/bin/sspl_setup

```diff
@@ -1,97 +1,234 @@
-# cortx-questions@seagate.com.
-
-set -e -u -o pipefail
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
 
 ######################################################################
 # This script serves as an entry point for various operations for SSPL
 # as needed by provisioner.
 ######################################################################
 
-SCRIPT_NAME=$(basename $0)
-SCRIPT_DIR=$(dirname $0)
-source $SCRIPT_DIR/constants.sh
-
-SSPL_CONFIGURED="/var/$PRODUCT_FAMILY/sspl/sspl-configured"
-
-usage() {
-    cat << EOF
-$SCRIPT_NAME [post_install[-p <LDR_R1>]|init [-dp] [-r <ssu|gw|cmu|vm>]|config [-f] [-r <ssu|gw|cmu|vm>]|test [self|sanity]|reset [hard -p <LDR_R1>|soft]]
-init options:
-    -dp Create configured datapath
-    -r  Role to be configured on the current node
-config options:
-    -f  Force reinitialization. Do not prompt
-    -r  Role to be configured on the current node
-post_install options:
-    -p Product to be configured
-reset options:
-    -p product to be configured
-EOF
-    exit 1
-}
-
-cmd=
-[ $# -ge 1 ] && cmd=$1 && shift 1
-
-case $cmd in
-    post_install )
-        $SCRIPT_DIR/sspl_post_install $*
-        exit $?
-        ;;
-
-    init )
-        $SCRIPT_DIR/sspl_setup_init $*
-        exit $?
-        ;;
-
-    config )
-        $SCRIPT_DIR/sspl_config $*
-        exit $?
-        ;;
-
-    test )
-        $SCRIPT_DIR/sspl_test $*
-        exit $?
-        ;;
-
-    support_bundle )
-        $SCRIPT_DIR/sspl_bundle_generate $*
-        exit $?
-        ;;
-
-    manifest_support_bundle )
-        $SCRIPT_DIR/manifest_support_bundle $*
-        exit $?
-        ;;
-
-    reset )
-        $SCRIPT_DIR/sspl_reset $*
-        exit $?
-        ;;
-
-    check )
-        sudo python3 $SCRIPT_DIR/validate_consul_config.py
-        [ -f $SSPL_CONFIGURED ] && exit $?
-        logger -i -p local3.err "SSPL is not configured. Run provisioner scripts in $SSPL_BASE_DIR/bin."
-        exit 1
-        ;;
-
-    * )
-        usage
-        ;;
-esac
+import sys
+import errno
+import argparse
+import inspect
+import traceback
+import os
+import subprocess
+# Add the top level directories
+sys.path.insert(0, '/opt/seagate/cortx/sspl/low-level')
```

Comment:
this way of adding to path is spreading everywhere now.

Reply:
We are addressing this through https://jts.seagate.com/browse/EOS-16240